subprojects/core/src/integTest/groovy/org/gradle/api/tasks/AbstractLineEndingSensitivityIntegrationSpec.groovy

```diff
@@ -189,6 +189,94 @@ abstract class AbstractLineEndingSensitivityIntegrationSpec extends AbstractInte
         api << Api.values()
     }
 
+    def "runtime classpath properties with zip entries can ignore line endings when specified (#api)"() {
+        createTaskWithNormalization(Classpath, LineEndingSensitivity.NORMALIZE_LINE_ENDINGS, null, api)
+
+        buildFile << """
+            task jarTask(type: Jar) {
+                from(project.files("foo"))
+                archiveFile.set(project.file("\${buildDir}/foo.jar"))
+            }
+            taskWithInputs {
+                sources.from(jarTask)
+                outputFile = project.file("\${buildDir}/output.txt")
+            }
+        """
+        file('foo/Changing.txt') << toUnix(TEXT_WITH_LINE_ENDINGS)
+        file('foo/Changing.class').bytes = CLASS_FILE_WITH_LF
+
+        when:
+        execute("taskWithInputs")
+
+        then:
+        executedAndNotSkipped(":taskWithInputs")
+
+        when:
+        file('foo/Changing.txt').text = toWindows(TEXT_WITH_LINE_ENDINGS)
+        cleanWorkspace()
+        execute("taskWithInputs")
+
+        then:
+        reused(":taskWithInputs")
+
+        when:
+        file('foo/Changing.class').bytes = CLASS_FILE_WITH_CRLF
```

Comment:
Heh...I never considered the actual likelihood of a change like this, but for the purposes of the test it's really just to prove that for a binary file (any binary file) a change that only differs by line endings will not result in a false positive cache hit.  Having said that, I changed this to jpg content, although I'm not sure that's any more realistic.

Reply:
JPEG is fine, though we could just say "binary" as well. Some tests are still referring to `CLASS_FILE_WITH_CRLF`, is that intentional?

Reply:
That was just the compile classpath test which is intended to prove that it never uses line ending normalization.  To your point, though, we would never have a class file with only a line ending difference, so that test wasn't testing anything realistic anyways.  I've just removed it and removed the class file content from the test.